### PR TITLE
feat(build): Mount local build context into the run command

### LINKED ISSUE #811: feat(build): Use build context instead of copy for conda-env.yml
## Description

Now we have to play tricks with host during the build process. We should be able to support reading these host files from the build context.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### LINKED ISSUE #813: bug(run): Failed to run ["cd xx", "ls"]
## Additional Info

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by gaocegege**
Fix #811 
Fix #813 

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

**Comment by aseaday** [comment]
Is it still in Work?

**Comment by gaocegege** [comment]
Yep

**Review by gaocegege — COMMENTED** [review]
/assign @aseaday @kemingy @VoVAllen

File: pkg/lang/ir/system.go [hunk 1]
```diff
@@ -55,10 +55,23 @@ func (g Graph) compileRun(root llb.State) llb.State {
 		return root.Run(llb.Shlex(fmt.Sprintf("bash -c \"%s\"", g.Exec[0]))).Root()
 	}
 
-	run := root.Run(llb.Shlex(fmt.Sprintf("bash -c \"%s\"", g.Exec[0])))
-	for _, c := range g.Exec[1:] {
-		run = run.Run(llb.Shlex(fmt.Sprintf("bash -c \"%s\"", c)))
+	var sb strings.Builder
+	sb.WriteString("set -e\n")
```
Comment (kemingy) [inline pkg/lang/ir/system.go:59]:
```suggestion
	sb.WriteString("set -euo pipefail\n")
```

File: pkg/lang/ir/system.go [hunk 2]
```diff
@@ -55,10 +55,23 @@ func (g Graph) compileRun(root llb.State) llb.State {
...
+		sb.WriteString(c + "\n")
 	}
+
+	cmdStr := fmt.Sprintf("bash -c '%s'", sb.String())
+	logrus.WithField("command", cmdStr).Debug("compile run command")
+	workingDir := g.getWorkingDir()
+	run := root.Dir(workingDir).
+		Run(llb.Shlex(cmdStr))
+	// Mount the build context into the build process.
+	// TODO(gaocegege): Maybe we should make it readonly, '
```
Comment (kemingy) [inline pkg/lang/ir/system.go:70]:
```suggestion
	// TODO(gaocegege): Maybe we should make it readonly,
```

**Review by kemingy — APPROVED** [review]

**Pull request merged**